### DISCUSSION TITLE
anyway to connect RethinkDB?

### DISCUSSION THREAD

**Discussion by sdekna** [Budibase/budibase#6617] [Help]
Hi there,
Is there any way where I can connect my app to my rethinkdb? maybe in javascript or something?

**Comment by melohagan** [comment]
Hi @sdekna 

Firstly, there is no pub/sub, websockets, realtime functionality within Budibase at the moment.

With that said, you may be able to use the RethinkDB API as a more traditional CRUD data source.

There is a guide here for contributing a new data connector to the Budibase codebase if that's something you would be comfortable with:
https://docs.budibase.com/docs/writing-your-own-external-data-connector
